src/odemis/util/spot.py

```diff
@@ -126,86 +128,260 @@ def SpotIntensity(data, background=None):
+
+    Parameters
+    ----------
+    r : Non-negative integer. Disk radius (default: 3)
+
+    Returns
+    -------
+    nhood : Structuring element
+    """
+    y, x = np.mgrid[-r:r + 1, -r:r + 1]
+    nhood = (x * x + y * y) <= (r * (r + 1))
+    return nhood
+
+
+def MaximaFind(image, qty, nsize=18):
+    """
+
+    Parameters
+    ----------
+    image : array like
+        Data array containing the greyscale image.
+    qty : int
+        The amount of maxima you want to find.
+    nsize : int
+
+    Returns
+    -------
+    refined_position : array like
+        A 2D array of shape (N, 2) containing the coordinates of the maxima.
+
+    """
+    filtered = BandPassFilter(image, 1, nsize)
+
+    # dilation
+    s = sedisk(nsize // 2)
+    dilated = ndimage.grey_dilation(filtered, footprint=s)
+
+    # find local maxima
+    binary = (filtered == dilated)
+
+    # thresholding
+    qradius = max(nsize // 4, 1)
+    BWdil = ndimage.binary_dilation(binary, structure=sedisk(qradius))
+
+    labels, num_features = ndimage.label(BWdil)
+    index = np.arange(1, num_features + 1)
+    mean = ndimage.mean(filtered, labels=labels, index=index)
+    si = np.argsort(mean)[::-1][:qty]
+    # estimate spot center position
+    pos = np.array(ndimage.center_of_mass(filtered, labels=labels,
+                                          index=index[si]))[:, ::-1]
+    if np.any(np.isnan(pos)):
+        logging.warning('not all positions found')
+        pos = pos[np.any(~np.isnan(pos), axis=1)]
+    # improve center estimate using radial symmetry method
+    w = nsize // 2
+    refined_position = np.zeros_like(pos)
+    for idx, xy in enumerate(pos):
+        _ij = np.rint(xy).astype(int)
+        i0, j0 = _ij - w + 1
+        i1, j1 = _ij + w
+        # if spot is close to edge of image i0 and j0 can be smaller than 0, crop so the spot is still in the center.
+        if i0 < 0:
+            i1, j1 = np.array([i1, j1]) + i0
+            i0, j0 = np.array([i0, j0]) - i0
+        if j0 < 0:
+            i1, j1 = np.array([i1, j1]) + j0
+            i0, j0 = np.array([i0, j0]) - j0
+        spot = filtered[j0:j1, i0:i1]
+        refined_position[idx] = np.rint(xy) + np.array(FindCenterCoordinates(spot))
+
+    return refined_position
+
+
+def EstimateLatticeConstant(pos):
+    """
+    Estimate the lattice constant of a point set that represent a square grid.
+
+    Parameters
+    ----------
+    pos : array like
+        A 2D array of shape (N, 2) containing the coordinates of the points.
+
+    Returns
+    -------
+    kxy : lattice constants
+    """
+    # Find the closest 4 neighbours (excluding itself) for each point.
+    tree = KDTree(pos)
+    dd, ii = tree.query(pos, k=5)
+    dr = dd[:, 1:]
+
+    # Determine the median radial distance and filter all points beyond
+    # 2*sigma.
+    med = np.median(dr)
+    std = np.std(dr)
+    outliers = np.abs(dr - med) > 2. * std  # doesn't work well if std is very high
```

Comment:
Can you add parenthesis to help read that line? I guess it's: `np.abs(dr - med) > (2 * std)`?

Also, the `.` in the integers of this function should not be needed in most of the cases. Could you try to remove the one not needed?